### PR TITLE
[wip] Span v2 datamodel

### PR DESCRIPTION
**Pull request by fedj**

### CODE REVIEW

File: Src/zipkin4net/Src/Internal/V2SpanConverter.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                            remoteEndpointType = zipkinCoreConstants.CLIENT_ADDR;
+                            kind = V2.Span.SpanKind.SERVER;
+                            break;
+                        case zipkinCoreConstants.CLIENT_RECV:
+                            kind = V2.Span.SpanKind.CLIENT;
+                            cr = v1Annotation;
+                            break;
+                        case zipkinCoreConstants.MESSAGE_SEND:
+                            kind = V2.Span.SpanKind.PRODUCER;
+                            ms = v1Annotation;
```
Comment (codefromthecrypt) [inline Src/zipkin4net/Src/Internal/V2SpanConverter.cs:62]: remoteEndpointType = zipkinCoreConstants.MESSAGE_ADDR;

Comment (fedj) [reply Src/zipkin4net/Src/Internal/V2SpanConverter.cs:62]: I agree, it should also be corrected here: https://github.com/openzipkin/zipkin/blob/master/zipkin/src/main/java/zipkin/internal/V2SpanConverter.java#L384

**Comment by codefromthecrypt** [comment]
@jcarres-mdsol v2 model is important as bugs are reported in this model but instrumentation doesnt emit it. imho this is a critical change especially to remove doubt from support on conversion related concerns

**Comment by jcarres-mdsol** [comment]
12K new lines is a pretty massive change 
Also seems pieces of this have made it into master already #194 
It seems there was a rebase 3 weeks ago. Can you tell us about the status of this @fedj  ?

**Comment by codefromthecrypt** [comment]
closed because it is intimidating for maintainers to look at this amount of change to try and recover it. The branch still exists in case they decide to!

**Pull request closed without merging**